tests/scripts/spack_manifest/test_getter.py

```diff
@@ -0,0 +1,480 @@
+            "spack": {
+                "specs": ["access-om2@git.1.0.0"]
+            }
+        }
+
+        root_spec_getter = RootSpec(manifest)
+        name = root_spec_getter.get_name()
+
+        assert name == "access-om2", "Root spec name should be correctly extracted."
+
+    def test_get_name__invalid(self):
+        manifest = {
+            "spack": {
+                "specs": ["@git.1.0.0"]
+            }
+        }
+
+        root_spec_getter = RootSpec(manifest)
+        with pytest.raises(NoSectionComponentError):
+            root_spec_getter.get_name()
+
+    def test_get_ref__valid_git(self):
+        manifest = {
+            "spack": {
+                "specs": ["access-om2@git.1.0.0"]
+            }
+        }
+
+        root_spec_getter = RootSpec(manifest)
+        ref = root_spec_getter.get_ref()
+
+        assert ref == "1.0.0", "Root spec ref should be correctly extracted."
+
+    def test_get_ref__valid_non_git(self):
+        manifest = {
+            "spack": {
+                "specs": ["access-om2@1.0.0"]
+            }
+        }
+
+        root_spec_getter = RootSpec(manifest)
+        ref = root_spec_getter.get_ref()
+
+        assert ref == "1.0.0", "Root spec ref should be correctly extracted."
+
+    def test_get_ref__valid_git_with_constraints(self):
+        manifest = {
+            "spack": {
+                "specs": ["access-om2@git.1.0.0 %compiler@2.0.0"]
+            }
+        }
+
+        root_spec_getter = RootSpec(manifest)
+        ref = root_spec_getter.get_ref()
+
+        assert ref == "1.0.0", "Root spec ref should be correctly extracted with constraints."
+
+    def test_get_ref__valid_git_with_space_constraints(self):
+        manifest = {
+            "spack": {
+                "specs": ["access-om2@git.1.0.0 %compiler@2.0.0 +debug"]
+            }
+        }
+
+        root_spec_getter = RootSpec(manifest)
+        ref = root_spec_getter.get_ref()
+
+        assert ref == "1.0.0", "Root spec ref should be correctly extracted with space constraints."
+
+    def test_get_ref__valid_git_with_spack_version_and_constraints(self):
+        manifest = {
+            "spack": {
+                "specs": ["access-om2@git.1.0.0=develop %compiler@2.0.0 +debug"]
+            }
+        }
+
+        root_spec_getter = RootSpec(manifest)
+        ref = root_spec_getter.get_ref()
+
+        assert ref == "1.0.0", "Root spec ref should be correctly extracted with Spack version and constraints."
+
+    def test_get_ref__invalid(self):
+        manifest = {
+            "spack": {
+                "specs": ["access-om2"]
+            }
+        }
+
+        root_spec_getter = RootSpec(manifest)
+        with pytest.raises(NoSectionComponentError):
+            root_spec_getter.get_ref()
+
+    @pytest.mark.parametrize(
+        "root_spec",
+        [
+            "",
+            "%compiler@2.0.0",
+            " %compiler@2.0.0 +debug",
+            "+debug",
+            " ~debug",
```

Comment:
Is the whitespace required for `~debug`? If not it should be straightforward to add a couple of extra tests, e.g.

```suggestion
            "+debug",
            "~debug",
            " +debug",
            " ~debug",
```
Or perhaps these are effectively redundant?